projects/igniteui-angular/src/lib/directives/input/input.directive.spec.ts

```diff
@@ -227,13 +226,13 @@ describe('IgxInput', () => {
         fix.componentInstance.model.firstName = 'Mike';
         fix.detectChanges();
 
-        fix.whenStable().then(() => {
-            fix.detectChanges();
-            igxInputGroups = fix.debugElement.queryAll(By.css('igx-input-group'));
-            igxInputGroups.forEach(element => {
-                const inputGroup = element.nativeElement;
-                expect(inputGroup.classList.contains('igx-input-group--filled')).toBe(true);
-            });
+        tick();
+        fix.detectChanges();
```

Comment:
There is no need to invoke `detectChanges` twice. You could remove this or the one above the `tick`.

Reply:
This one should remain, because some of the styles checked need time to be applied.